### ISSUE TITLE
Remove deprecated t.conch.ssh.userauth.SSHUserAuthClient's getPublicKey and getPrivateKey

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4106]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#4106|
|Type|task|
|Created|2009-11-13 20:29:42Z|

```getPublicKey``` and ```getPrivateKey``` in [twisted/conch/ssh/userauth.py](twisted/conch/ssh/userauth.py) are deprecated since Twisted 9.0 so let's remove them.

<details><summary>Searchable metadata</summary>

```
trac-id__4106 4106
type__task task
reporter__thijs thijs
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__question question
time__1258144182000000 1258144182000000
changetime__1503800173744789 1503800173744789
version__None None
owner__ 
cc__thijs cc__exarkun
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_1"></a>|@cyli set owner to @thijstriemstra|
|-|-|

It doesn't seem as if getPublicKey and getPrivateKey are deprecated.  What is deprecated is the code supporting those methods returning a string value, rather than a `twisted.conch.ssh.keys.Key`.  Should that code really be removed right now, considering that there were a lot of other previously-deprecated removals in [twisted/conch/ssh/keys.py](twisted/conch/ssh/keys.py) already?  This doesn't seem to be an outward-facing api change.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_2"></a>|@cyli commented|
|-|-|

Or rather, the outward facing API change would be the return value of getPublicKey and getPrivateKey - the methods themselves aren't deprecated.  The code that handles the legacy string-return-value code is not an outward-facing API change.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_6"></a>|@rodrigc set status to `closed`|
|-|-|

Fixed in [ticket:7054] in In changeset 9570a8f4a8141592910e395d8a3b32bfd8ad463e
